src/metabase/query_processor/middleware/optimize_temporal_filters.clj

```diff
@@ -0,0 +1,205 @@
+  (mbql.u/replace field
+    [:datetime-field wrapped _]
+    [:datetime-field wrapped :default]))
+
+(defmulti ^:private temporal-value-lower-bound
+  "Get a clause representing the *lower* bound that should be used when converting a `temporal-value-clause` (e.g.
+  `:absolute-datetime` or `:relative-datetime`) to an optimized range."
+  {:arglists '([temporal-value-clause datetime-field-unit])}
+  mbql.u/dispatch-by-clause-name-or-class)
+
+(defmulti ^:private temporal-value-upper-bound
+  "Get a clause representing the *upper* bound that should be used when converting a `temporal-value-clause` (e.g.
+  `:absolute-datetime` or `:relative-datetime`) to an optimized range."
+  {:arglists '([temporal-value-clause datetime-field-unit])}
+  mbql.u/dispatch-by-clause-name-or-class)
+
+(defmethod temporal-value-lower-bound :absolute-datetime
+  [[_ t unit] _]
+  [:absolute-datetime (temporal-literal-lower-bound unit t) :default])
+
+(defmethod temporal-value-upper-bound :absolute-datetime
+  [[_ t unit] _]
+  [:absolute-datetime (temporal-literal-upper-bound unit t) :default])
+
+(defmethod temporal-value-lower-bound :relative-datetime
+  [[_ n unit] datetime-field-unit]
+  [:relative-datetime (if (= n :current) 0 n) (or unit datetime-field-unit)])
+
+(defmethod temporal-value-upper-bound :relative-datetime
+  [[_ n unit] datetime-field-unit]
+  [:relative-datetime (inc (if (= n :current) 0 n)) (or unit datetime-field-unit)])
+
+(defmulti ^:private optimize-filter
+  "Optimize a filter clause agat a bucketed `:datetime-field` clause and `:absolute-datetime` or `:relative-datetime`
+  value by converting to an unbucketed range."
+  {:arglists '([clause])}
+  mbql.u/dispatch-by-clause-name-or-class)
+
+(defmethod optimize-filter :=
+  [[_ field temporal-value]]
+  (let [[_ _ datetime-field-unit] (mbql.u/match-one field :datetime-field)]
+    (when (field-and-temporal-value-have-compatible-units? field temporal-value)
+      (let [field' (change-datetime-field-unit-to-default field)]
+        [:and
+         [:>= field' (temporal-value-lower-bound temporal-value datetime-field-unit)]
+         [:< field'  (temporal-value-upper-bound temporal-value datetime-field-unit)]]))))
+
+(defmethod optimize-filter :!=
+  [filter-clause]
+  (mbql.u/negate-filter-clause ((get-method optimize-filter :=) filter-clause)))
+
+(defn- optimize-comparison-filter
+  [optimize-temporal-value-fn [filter-type field temporal-value] new-filter-type]
+  [new-filter-type
+   (change-datetime-field-unit-to-default field)
+   (optimize-temporal-value-fn temporal-value (datetime-field-unit field))])
+
+(defmethod optimize-filter :<
+  [filter-clause]
+  (optimize-comparison-filter temporal-value-lower-bound filter-clause :<))
+
+(defmethod optimize-filter :<=
+  [filter-clause]
+  (optimize-comparison-filter temporal-value-upper-bound filter-clause :<))
+
+(defmethod optimize-filter :>
+  [filter-clause]
+  (optimize-comparison-filter temporal-value-upper-bound filter-clause :>=))
```

Comment:
Similar to above.  It's passing `:>=` through from the `:>` case.

Reply:
yup, it's intentional